sound/soc/sof/sof-audio.c

```diff
@@ -85,6 +88,8 @@ int sof_widget_setup(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
 	if (!swidget->private)
 		return 0;
 
+	trace_sof_widget_setup(swidget);
```

Comment:
You probably need to explain how a user might enable such specific tracepoints?

I am not sure this is usable at all, maybe you need to move this at a higher level in the loops that start setting up all the widgets or freeing them.

In other words, I don't see much value in tracing the state of individual widgets?

@ranj063 would you concur?

Reply:
@plbossart I think the effect will be the same irrespective of whether we place this in the loop outside or the individual sof_widget_setup() call. I think this is a better place to trace and tracing the use_count per widget is quite useful. We've been burned with unbalanced use_counts in the past during suspend/resume and this will offer us the clues as to what is going on. If you're thinking along the lines of tracing when widgets are getting setup/freed, we probably need another tracepoint for it, I think